exchanges/okx/okx_wrapper.go

```diff
@@ -0,0 +1,1218 @@
+	}
+	return acc, nil
+}
+
+// GetFundingHistory returns funding history, deposits and withdrawals
+func (ok *Okx) GetFundingHistory(ctx context.Context) ([]exchange.FundHistory, error) {
+	depositHistories, er := ok.GetCurrencyDepositHistory(ctx, "", "", "", -1, time.Time{}, time.Time{}, 0)
+	if er != nil {
+		return nil, er
+	}
+	withdrawalHistories, er := ok.GetWithdrawalHistory(ctx, "", "", "", "", -5, time.Time{}, time.Time{}, 0)
+	if er != nil {
+		return nil, er
+	}
+	resp := make([]exchange.FundHistory, 0, len(depositHistories)+len(withdrawalHistories))
+	for x := range depositHistories {
+		resp = append(resp, exchange.FundHistory{
+			ExchangeName:    ok.Name,
+			Status:          strconv.Itoa(depositHistories[x].State),
+			Timestamp:       depositHistories[x].Timestamp,
+			Currency:        depositHistories[x].Currency,
+			Amount:          depositHistories[x].Amount,
+			TransferType:    "deposit",
+			CryptoToAddress: depositHistories[x].ToDepositAddress,
+			CryptoTxID:      depositHistories[x].TransactionID,
+		})
+	}
+	for x := range withdrawalHistories {
+		resp = append(resp, exchange.FundHistory{
+			ExchangeName:    ok.Name,
+			Status:          withdrawalHistories[x].StateOfWithdrawal,
+			Timestamp:       withdrawalHistories[x].Timestamp,
+			Currency:        withdrawalHistories[x].Currency,
+			Amount:          withdrawalHistories[x].Amount,
+			TransferType:    "withdrawal",
+			CryptoToAddress: withdrawalHistories[x].ToReceivingAddress,
+			CryptoTxID:      withdrawalHistories[x].TransactionID,
+			TransferID:      withdrawalHistories[x].WithdrawalID,
+			Fee:             withdrawalHistories[x].WithdrawalFee,
+			CryptoChain:     withdrawalHistories[x].ChainName,
+		})
+	}
+	return resp, nil
+}
+
+// GetWithdrawalsHistory returns previous withdrawals data
+func (ok *Okx) GetWithdrawalsHistory(ctx context.Context, c currency.Code, _ asset.Item) (resp []exchange.WithdrawalHistory, err error) {
+	withdrawals, er := ok.GetWithdrawalHistory(ctx, c.String(), "", "", "", -5, time.Time{}, time.Time{}, 0)
+	if er != nil {
+		return nil, er
+	}
+	resp = make([]exchange.WithdrawalHistory, 0, len(withdrawals))
+	for x := range withdrawals {
+		resp = append(resp, exchange.WithdrawalHistory{
+			Status:          withdrawals[x].StateOfWithdrawal,
+			Timestamp:       withdrawals[x].Timestamp,
+			Currency:        withdrawals[x].Currency,
+			Amount:          withdrawals[x].Amount,
+			TransferType:    "withdrawal",
+			CryptoToAddress: withdrawals[x].ToReceivingAddress,
+			CryptoTxID:      withdrawals[x].TransactionID,
+			CryptoChain:     withdrawals[x].ChainName,
+			TransferID:      withdrawals[x].WithdrawalID,
+			Fee:             withdrawals[x].WithdrawalFee,
+		})
+	}
+	return resp, nil
+}
+
+// GetRecentTrades returns the most recent trades for a currency and asset
+func (ok *Okx) GetRecentTrades(ctx context.Context, p currency.Pair, assetType asset.Item) ([]trade.Data, error) {
+	const limit = 1000
```

Comment:
There are a few declared consts for limit can be a global and remove all references in functions.